xinference/model/core.py

```diff
@@ -12,4 +12,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: define a data structure for LLMs, Speech recognition models, and etc.
+from abc import ABC
+from typing import Any, Optional, Tuple
+
+
+class ModelDescription(ABC):
+    def to_description(self):
```

Comment:
1. `to_dict` may be a more readable name.
2. better add type hint for the return value.